### PR TITLE
Upcoming events card

### PR DESCRIPTION
**Pull request by Tramsi**

### CODE REVIEW

**Comment by ahmedsamir96m** [comment]
kindly consider the conflict in App.js

**Comment by elhaw** [comment]
> kindly consider the conflict in App.js

I think it has been resolved 👍  you can review it again

File: src/components/UpcomingEventsCard/index.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        this.setState({
+          loading: false,
+          error: true,
+          errorMessage: "couldn't fetch data"
+        });
+      });
+  };
+
+  componentDidMount() {
+    this._getData();
```
Comment (Nilegfx) [inline src/components/UpcomingEventsCard/index.jsx:35]: I liked how you separated your data fetching in a separate function and cleaned up your react lifecycle method.

File: src/components/UpcomingEventsCard/index.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  progress = (raised, goal) => {
```
Comment (Nilegfx) [inline src/components/UpcomingEventsCard/index.jsx:38]: try to be more descriptive here, for example, `progressPrecentage`

File: src/components/UpcomingEventsCard/index.jsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                <div className="number">
+                  <h2>
+                    {this.progress(
+                      this.state.data.featured_cause.raised,
+                      this.state.data.featured_cause.Goal
+                    )}
+                    %
+                  </h2>
+                </div>
+                {/* <svg className="dot">
```
Comment (Nilegfx) [inline src/components/UpcomingEventsCard/index.jsx:92]: is this suppose to be the circle in the design?

Comment (Tramsi) [reply src/components/UpcomingEventsCard/index.jsx:92]: Yup .. but I couldn't figure how to make it dynamic with the progress bar.

Comment (elhaw) [reply src/components/UpcomingEventsCard/index.jsx:92]: **_Yes_** this is the small circle. 
We had a challenge in how we make this circle moves when the percentage value of progress changes.
Capture

**Our approach**
-  
We can make the position of small circle `(top` and `left` property) related to the percentage
of progress value.

**Or**

We can make it fixed.

**Pull request merged**